internal/projection/process/event_enricher_test.go

```diff
@@ -302,6 +302,58 @@ var _ = Describe("Process message", func() {
 			assertHostsSummary(enrichedEvent)
 		})
 	})
+	When("Enrich a cluster event with different release tag locations", func() {
+		It("with release tag as metadata", func() {
+			myMetadata := map[string]interface{}{
+				"versions": map[string]interface{}{
+					"release_tag": "foobar",
+					"versions":    `{"assisted-installer": "registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-rhel8:latest", "assisted-installer-controller": "registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-reporter-rhel8:latest", "assisted-installer-service": "quay.io/app-sre/assisted-service:880ad10", "discovery-agent": "registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-agent-rhel8:latest"}`,
```

Comment:
I think I have mislead you with the code snippets...
This is the *raw* payload we get from kafka:
```
  "Metadata": {
    "versions": {
      "release_tag": "master",
      "versions": {
        "assisted-installer": "registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-rhel8:latest",
        "assisted-installer-controller": "registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-reporter-rhel8:latest",
        "assisted-installer-service": "quay.io/app-sre/assisted-service:880ad10",
        "discovery-agent": "registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-agent-rhel8:latest"
      }
    }
  }
```
It will be decoded into https://github.com/openshift-assisted/assisted-events-stream/blob/master/internal/types/event.go#L6, therefore it would be already structured in a `map[string]interface{}`. I don't think we need to handle the case where this is actually double JSON encoded.